src/components/Navbar.js

```diff
@@ -12,43 +9,63 @@ const argumentsSet = [
   ["Investors", "For investors"]
 ];
 
+//WARNING: the wheel was invented here!!!
 class Navbar extends React.Component {
   constructor(props) {
     super(props);
     this.state = {activeLinkName: argumentsSet[0][0]};
   }
+  componentDidMount() {
+    window.addEventListener("scroll", () => this.onWindowScroll(), true);
+  }
+  onWindowScroll() {
```

Comment:
[2/2 - listener] Please use `onWindowScroll = () => {` 

why?

because it will add property to the class (of React component) at runtime. Property assign will be after babel transpile and es6 things like this (pseudocode):
```js
class NavBar {
  constructor {
    this. onWindowScroll = () => { /* source code for  onWindowScroll method*/ }
    // yup, arrow function. therefore, it will have be binded lexically to current class instance of class NavBar
  }
}
```